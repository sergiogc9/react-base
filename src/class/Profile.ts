export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
