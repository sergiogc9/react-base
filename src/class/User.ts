export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
