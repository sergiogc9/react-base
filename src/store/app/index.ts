import { all } from 'redux-saga/effects';
import { combineReducers } from 'redux';

import * as auth from './auth';
import * as notifications from './notifications';
import * as profile from './profile';

// define sagas generator function to register all sagas
export function* sagas() {
	yield all([
		auth.sagas(),
		profile.sagas(),
		notifications.sagas()
	]);
}

export const reducers = combineReducers({
	auth: auth.reducers,
	profile: profile.reducers,
	notifications: notifications.reducers
});

export interface State {
	auth: auth.State;
	profile: profile.State;
	notifications: notifications.State;
}

export const INITIAL_STATE = {
	auth: auth.INITIAL_STATE,
	profile: profile.INITIAL_STATE,
	notifications: notifications.INITIAL_STATE
};
