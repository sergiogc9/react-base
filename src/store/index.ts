import { createBrowserHistory } from 'history';
import isEqual from 'lodash/isEqual';
import { combineReducers, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { all } from 'redux-saga/effects';

import { connectRouter } from 'connected-react-router';

import { connectApi } from '@src/lib/ajax/Api';
import { getMiddleware, sagaMiddleware } from '@src/middleware';
import { State } from './types';

import * as app from './app';

const INITIAL_STATE: State = {
	app: app.INITIAL_STATE,
	router: { action: "POP", location: { hash: "", pathname: "/", search: "", state: undefined } }
};

// define sagas generator function to register all sagas
function* sagas() {
	yield all([
		app.sagas()
	]);
}

const history = createBrowserHistory();
// patch to avoid stacking same location multiple times in history
const historyPush = history.push;
let lastLocation = history.location;
history.listen(location => lastLocation = location);
history.push = (path: any, state: object = {}): void => {
	const lastPath = lastLocation.pathname + lastLocation.hash + lastLocation.search;
	if (lastLocation === null || path !== lastPath || !isEqual(state, lastLocation.state)) historyPush(path, state);
};

// create reducers

const reducers = combineReducers({
	app: app.reducers,
	router: connectRouter(history)
});

const store = createStore(reducers, composeWithDevTools(getMiddleware(history)));
connectApi(store);

sagaMiddleware.run(sagas);

export { INITIAL_STATE, sagas, history, reducers, store };
