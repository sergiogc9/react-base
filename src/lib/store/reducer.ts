import { CaseReducer, createSlice, EntityState, PayloadAction } from '@reduxjs/toolkit';

import { EntitiesFetchData } from 'store/entities';

// General reducer stuff
type CaseReducerWithPrepare<State, Payload> = {
	reducer: CaseReducer<State, {
		payload: Payload;
		type: string;
	}>;
	prepare: (payload: Payload) => {
		payload: Payload;
		meta: object | undefined;
	};
}

const getReducer = <State, Payload = void>(reducer: CaseReducer<State, PayloadAction<Payload>>, meta?: object): CaseReducerWithPrepare<State, Payload> => ({
	reducer,
	prepare: (payload: Payload) => ({ payload, meta })
});

// Api specific reducer stuff
export type DefaultPayloadError = { code: string, message: string };

type ApiOperatorOptions = {
	reload?: boolean // Used to reload page when optimistic api call fails
};

type ApiReducers<State, StartPayload, SuccessPayload, ErrorPayload> = {
	start?: CaseReducer<State, PayloadAction<StartPayload>>
	success?: CaseReducer<State, PayloadAction<SuccessPayload>>
	error?: CaseReducer<State, PayloadAction<ErrorPayload>>
}

const getApiReducers = <State, StartPayload = void, SuccessPayload = void, ErrorPayload = DefaultPayloadError>(
	reducers: ApiReducers<State, StartPayload, SuccessPayload, ErrorPayload>,
	options?: ApiOperatorOptions
): [
		CaseReducerWithPrepare<State, StartPayload>,
		CaseReducerWithPrepare<State, SuccessPayload>,
		CaseReducerWithPrepare<State, ErrorPayload>
	] => {
	const defaultReducer: CaseReducer<State> = state => state as State;
	return [
		getReducer<State, StartPayload>(reducers.start || defaultReducer, { api: 'start' }),
		getReducer<State, SuccessPayload>(reducers.success || defaultReducer, { api: 'success' }),
		getReducer<State, ErrorPayload>(reducers.error || defaultReducer, { api: 'error', reload: options?.reload })
	];
};

const getEntityFetchReducers = <T, StartPayload = void, SuccessPayload = void, ErrorPayload = DefaultPayloadError>() => {
	return getApiReducers<EntityState<T> & EntitiesFetchData, StartPayload, SuccessPayload, ErrorPayload>({
		start: state => { state.status = 'loading'; },
		success: state => { state.status = 'loaded'; },
		error: state => { state.status = 'error'; }
	});
};

export { getReducer, getApiReducers, getEntityFetchReducers, createSlice };
