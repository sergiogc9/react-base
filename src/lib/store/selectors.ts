import { useSelector as reduxUseSelector } from 'react-redux';

import { State } from 'store/types';

type Selector<T> = (state :State) => T;

const useSelector = <T>(selector: Selector<T>) => reduxUseSelector<State, T>(selector);

export { useSelector };
