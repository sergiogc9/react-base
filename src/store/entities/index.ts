import { combineReducers } from '@reduxjs/toolkit';

import selectors from './selectors';
import * as author from './Author';
import * as book from './Book';

export const reducer = combineReducers({
	author: author.reducer,
	book: book.reducer
});

export interface State {
	author: typeof author.INITIAL_STATE;
	book: typeof book.INITIAL_STATE;
}

export const INITIAL_STATE = {
	author: author.INITIAL_STATE,
	book: book.INITIAL_STATE
};

export { selectors };
