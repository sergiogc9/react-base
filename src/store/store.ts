import { combineReducers, configureStore } from '@reduxjs/toolkit';

import { customMiddlewares, sagaMiddleware } from 'middleware';
import * as auth from './auth';
import * as entities from './entities';
import * as notifications from './notifications';
import * as ui from './ui';
import sagas from './sagas';

// create reducers
const reducers = combineReducers({
	auth: auth.reducer,
	entities: entities.reducer,
	notifications: notifications.reducer,
	ui: ui.reducer
});

const store = configureStore({
	reducer: reducers,
	middleware: getDefaultMiddleware => getDefaultMiddleware().concat(...customMiddlewares)
});
const INITIAL_STATE = store.getState();

sagaMiddleware.run(sagas);

export { reducers, store, INITIAL_STATE };
