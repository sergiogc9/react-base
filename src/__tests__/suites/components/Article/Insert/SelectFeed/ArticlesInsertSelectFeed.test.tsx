import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import merge from "lodash/merge";

import ArticleSelectFeed from "@src/components/Article/Insert/SelectFeed";
import { INITIAL_STATE } from "@src/store";
import { FeedObject } from "@src/class/Feed";
import { FocusFeeds, FocusObject } from "@src/class/Focus";
import TestHelper from '@src/__tests__/utils/Helper/TestHelper';

const document = TestHelper.getDocument();

const feedOnlineEnabled: FeedObject = {
	definition: {},
	deleted_at: new Date(1549287550563),
	enabled: true,
	filters: {},
	focus_id: "focus-id-1",
	id: "feed-id-1",
	inserted_at: new Date(1549287550563),
	name: "feed-name-1",
	oldest_document: document,
	type: "online",
	updated_at: new Date(1549287550563)
};

const feedOnlineDisabled: FeedObject = {
	...feedOnlineEnabled,
	enabled: false,
	id: "feed-id-2",
	name: "feed-name-2",
};

const feedSocialEnabled: FeedObject = {
	...feedOnlineEnabled,
	id: "feed-id-3",
	name: "feed-name-3",
	type: "socialmedia",
};

const feedSocialDisabled: FeedObject = {
	...feedOnlineEnabled,
	enabled: false,
	id: "feed-id-4",
	name: "feed-name-4",
	type: "socialmedia",
};

const feedPrintOld: FeedObject = {
	...feedOnlineEnabled,
	id: "feed-id-5",
	name: "feed-name-5",
	type: "print",
};

const feedPrintNew: FeedObject = {
	...feedOnlineEnabled,
	enabled: false,
	id: "feed-id-6",
	name: "feed-name-6",
	type: "print_percolator"
};

const focusFeeds: FocusFeeds = {
	online: [feedOnlineEnabled, feedOnlineDisabled],
	socialmedia: [feedSocialEnabled, feedSocialDisabled],
	print: [feedPrintOld, feedPrintOld]
};

const focus: FocusObject = {
	deleted_at: new Date(1549287550563),
	feeds: focusFeeds,
	id: "focus-id-1",
	inserted_at: new Date(1549287550563),
	name: "focus-name-1",
	oldest_document: document,
	updated_at: new Date(1549287550563),
	url_logo: "http://fake-url.com",
	acl_users: [
		{
			user_id: 'fake-user-id-1',
			discover_role: 'viewer',
			last_name: "Name",
			first_name: "User"
		},
		{
			user_id: 'fake-user-id-2',
			discover_role: 'viewer',
			last_name: 'User 2',
			first_name: 'Name 2'
		}
	]
};

const focus2: FocusObject = {
	...focus,
	feeds: { online: [], socialmedia: [], print: [] },
	id: "focus-id-2",
	name: "focus-name-2",
};

const focus3: FocusObject = {
	...focus,
	feeds: undefined,
	id: "focus-id-2",
	name: "focus-name-2",
};

const focusList: FocusObject[] = [focus, focus2, focus3];

const mockStore = configureMockStore();

function getWrappedComponent(component: JSX.Element, stateSlice: object) {
	const store = mockStore(merge({}, INITIAL_STATE, stateSlice));
	// Return also store to check actions executed
	return { store, component: mount(<Provider store={store}>{component}</Provider>) };
}

describe("<ArticleSelectFeed>", () => {
	let wrapper: any;

	beforeEach(() => {
		wrapper = getWrappedComponent(<ArticleSelectFeed />, {}).component;
	});

	it("Component by default", () => {
		expect(wrapper.html()).toMatchSnapshot();
		expect(wrapper.find('CircularProgress#insertDocumentDialogLoading')).toHaveLength(0);
	});

	it("Component by default with loading", () => {
		const { component } = getWrappedComponent(<ArticleSelectFeed />, {
			focus: { list: { focusList } },
			article: { insert: { loading: true } }
		});
		expect(component.html()).toMatchSnapshot();
		expect(component.find('CircularProgress#insertDocumentDialogLoading')).toHaveLength(1);
	});

	it("Component by default with focus list", () => {
		const { component } = getWrappedComponent(<ArticleSelectFeed />, {
			focus: { list: { focusList } }
		});
		expect(component.html()).toMatchSnapshot();
	});

	it("Component only with feed selected", () => {
		const { component } = getWrappedComponent(<ArticleSelectFeed />, {
			focus: { list: { focusList } },
			article: {
				insert: {
					selectedFeed: { id: feedOnlineEnabled.id, type: feedOnlineEnabled.type }
				}
			}
		});
		expect(component.html()).toMatchSnapshot();
		expect(component.find('input#insertDocumentDialogUrlInput')).toHaveLength(1);
	});

	it("Component only with print feed selected", () => {
		const { component } = getWrappedComponent(<ArticleSelectFeed />, {
			focus: { list: { focusList } },
			article: {
				insert: {
					selectedFeed: { id: feedPrintNew.id, type: feedPrintNew.type }
				}
			}
		});
		expect(component.html()).toMatchSnapshot();
		expect(component.find('input#insertDocumentDialogUrlInput')).toHaveLength(0);
	});

	it("Component only with invalid url", () => {
		const { component } = getWrappedComponent(<ArticleSelectFeed />, {
			focus: { list: { focusList } },
			article: {
				insert: {
					url: "badurl.bad.",
					invalidUrlError: "bad_error"
				}
			}
		});
		expect(component.html()).toMatchSnapshot();
		expect(component.find('label.md-text--error')).toHaveLength(1);
	});

	it("Select feed selected handler", () => {
		const { component, store } = getWrappedComponent(<ArticleSelectFeed />, {
			focus: { list: { focusList } },
			article: {
				insert: {
					selectedFeed: { id: feedOnlineEnabled.id, type: feedOnlineEnabled.type }
				}
			}
		});
		component.find('SelectField#insertDocumentDialogSelectFeed').at(0).props().onChange(feedPrintNew.id);
		expect(store.getActions()).toMatchObject([{ type: '@@article/insert/SET_SELECTED_FEED_ID', payload: { id: feedPrintNew.id } }]);
	});

	it("Select feed on blur url input", () => {
		const { component, store } = getWrappedComponent(<ArticleSelectFeed />, {
			focus: { list: { focusList } },
			article: {
				insert: {
					url: "gironafc.cat"
				}
			}
		});
		component.find('TextField#insertDocumentDialogUrlInput input').at(0).simulate('blur');
		expect(store.getActions()).toMatchObject([{ type: '@@article/insert/CHECK_URL', payload: {} }]);
	});

	it("Url text field change handler", () => {
		const { component, store } = getWrappedComponent(<ArticleSelectFeed />, {
			focus: { list: { focusList } },
			article: {
				insert: {
					selectedFeed: { id: feedOnlineEnabled.id, type: feedOnlineEnabled.type }
				}
			}
		});
		component.find('TextField#insertDocumentDialogUrlInput').at(0).props().onChange("http://gironafc.cat");
		expect(store.getActions()).toMatchObject([{ type: '@@article/insert/SET_URL', payload: { url: "http://gironafc.cat" } }]);
	});

	it("Url text field key down handler when next button is disabled", () => {
		const { component, store } = getWrappedComponent(<ArticleSelectFeed />, {
			focus: { list: { focusList } },
			article: {
				insert: {
					selectedFeed: { id: feedOnlineEnabled.id, type: feedOnlineEnabled.type }
				}
			}
		});
		component.find('TextField#insertDocumentDialogUrlInput').at(0).props().onKeyDown({ keyCode: 13 });
		expect(store.getActions()).toMatchObject([]);
	});

	it("Url text field key down handler with another key but enter", () => {
		const { component, store } = getWrappedComponent(<ArticleSelectFeed />, {
			focus: { list: { focusList } },
			article: {
				insert: {
					selectedFeed: { id: feedOnlineEnabled.id, type: feedOnlineEnabled.type },
					url: "http://gironafc.cat"
				}
			}
		});
		component.find('TextField#insertDocumentDialogUrlInput').at(0).props().onKeyDown({ keyCode: 2 });
		expect(store.getActions()).toMatchObject([]);
	});

	it("Url text field key down handler with enter key", () => {
		const { component, store } = getWrappedComponent(<ArticleSelectFeed />, {
			focus: { list: { focusList } },
			article: {
				insert: {
					selectedFeed: { id: feedOnlineEnabled.id, type: feedOnlineEnabled.type },
					url: "http://gironafc.cat"
				}
			}
		});
		component.find('TextField#insertDocumentDialogUrlInput').at(0).props().onKeyDown({ keyCode: 13 });
		expect(store.getActions()).toMatchObject([{ type: '@@article/insert/CHANGE_STEP', payload: { step: 'SELECT_FOUND_DOCUMENTS' }, meta: { api: 'start' } }]);
	});
});
