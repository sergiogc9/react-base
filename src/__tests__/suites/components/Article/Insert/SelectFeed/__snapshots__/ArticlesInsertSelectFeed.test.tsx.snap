// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleSelectFeed> Component by default 1`] = `"<div id=\\"articleInsertDocumentDialogSelectFeed\\"><div id=\\"insertDocumentDialogSelectFeed-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-full-width md-menu-container md-menu--select-field select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"insertDocumentDialogSelectFeed\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">feed.value</label><div id=\\"insertDocumentDialogSelectFeed-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\"></span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"insertDocumentDialogSelectFeed\\" value=\\"\\"></div></div></div><div class=\\"md-text-field-container md-full-width md-text-field-container--input text-field\\"><label for=\\"insertDocumentDialogUrlInput\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">Url</label><input id=\\"insertDocumentDialogUrlInput\\" type=\\"text\\" class=\\"md-text-field md-text-field--floating-margin md-full-width md-text\\" value=\\"\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"></div></div>"`;

exports[`<ArticleSelectFeed> Component by default with focus list 1`] = `"<div id=\\"articleInsertDocumentDialogSelectFeed\\"><div id=\\"insertDocumentDialogSelectFeed-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-full-width md-menu-container md-menu--select-field select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"insertDocumentDialogSelectFeed\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">feed.value</label><div id=\\"insertDocumentDialogSelectFeed-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\"></span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"insertDocumentDialogSelectFeed\\" value=\\"\\"></div></div></div><div class=\\"md-text-field-container md-full-width md-text-field-container--input text-field\\"><label for=\\"insertDocumentDialogUrlInput\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">Url</label><input id=\\"insertDocumentDialogUrlInput\\" type=\\"text\\" class=\\"md-text-field md-text-field--floating-margin md-full-width md-text\\" value=\\"\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"></div></div>"`;

exports[`<ArticleSelectFeed> Component by default with loading 1`] = `"<div id=\\"articleInsertDocumentDialogSelectFeed\\"><div id=\\"insertDocumentDialogSelectFeed-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-full-width md-menu-container md-menu--select-field select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"insertDocumentDialogSelectFeed\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">feed.value</label><div id=\\"insertDocumentDialogSelectFeed-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\"></span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"insertDocumentDialogSelectFeed\\" value=\\"\\"></div></div></div><div class=\\"md-text-field-container md-full-width md-text-field-container--input text-field\\"><label for=\\"insertDocumentDialogUrlInput\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">Url</label><input id=\\"insertDocumentDialogUrlInput\\" type=\\"text\\" class=\\"md-text-field md-text-field--floating-margin md-full-width md-text\\" value=\\"\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"></div></div><svg id=\\"insertDocumentDialogLoading\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"md-progress md-progress--circular md-block-centered md-progress--circular-indeterminate\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 66 66\\"><circle class=\\"md-circular-progress-path md-circular-progress-path--animated\\" stroke-width=\\"6\\" stroke-linecap=\\"round\\" cx=\\"33\\" cy=\\"33\\" r=\\"30\\"></circle></svg>"`;

exports[`<ArticleSelectFeed> Component only with feed selected 1`] = `"<div id=\\"articleInsertDocumentDialogSelectFeed\\"><div id=\\"insertDocumentDialogSelectFeed-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-full-width md-menu-container md-menu--select-field select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"insertDocumentDialogSelectFeed\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">feed.value</label><div id=\\"insertDocumentDialogSelectFeed-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">feed-name-1 (type.online)</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"insertDocumentDialogSelectFeed\\" value=\\"feed-id-1\\"></div></div></div><div class=\\"md-text-field-container md-full-width md-text-field-container--input text-field\\"><label for=\\"insertDocumentDialogUrlInput\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">Url</label><input id=\\"insertDocumentDialogUrlInput\\" type=\\"text\\" class=\\"md-text-field md-text-field--floating-margin md-full-width md-text\\" value=\\"\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"></div></div>"`;

exports[`<ArticleSelectFeed> Component only with invalid url 1`] = `"<div id=\\"articleInsertDocumentDialogSelectFeed\\"><div id=\\"insertDocumentDialogSelectFeed-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-full-width md-menu-container md-menu--select-field select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"insertDocumentDialogSelectFeed\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">feed.value</label><div id=\\"insertDocumentDialogSelectFeed-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\"></span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"insertDocumentDialogSelectFeed\\" value=\\"\\"></div></div></div><div class=\\"md-text-field-container md-full-width md-text-field-container--input text-field\\"><label for=\\"insertDocumentDialogUrlInput\\" class=\\"md-floating-label md-floating-label--floating md-text--error\\">Url</label><input id=\\"insertDocumentDialogUrlInput\\" type=\\"text\\" class=\\"md-text-field md-text-field--floating-margin md-full-width md-text\\" value=\\"badurl.bad.\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--text-field-error\\"><div class=\\"md-text-field-message-container md-full-width md-text--error\\"><div aria-hidden=\\"false\\" class=\\"md-text-field-message md-text-field-message--active\\">bad_error</div></div></div></div>"`;

exports[`<ArticleSelectFeed> Component only with print feed selected 1`] = `"<div id=\\"articleInsertDocumentDialogSelectFeed\\"><div id=\\"insertDocumentDialogSelectFeed-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-full-width md-menu-container md-menu--select-field select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"insertDocumentDialogSelectFeed\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">feed.value</label><div id=\\"insertDocumentDialogSelectFeed-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\"></span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"insertDocumentDialogSelectFeed\\" value=\\"feed-id-6\\"></div></div></div></div>"`;
